server/backend/TextToSpeech/YngpingTtsUtil.cs

```diff
@@ -77,15 +77,68 @@ public static class YngpingTtsUtil
                                              {"34", new string[]{"uai"}},
                                              {"36", new string[]{"eu"}}};
 
+        private static Dictionary<string, string> SyllableMappings =
+            new Dictionary<string, string>();
+
         // 松韵调 (会发生变韵的声调)
         private static HashSet<string> AltTones = new HashSet<string> { "03", "04", "07" };
         private static HashSet<string> AbruptTones = new HashSet<string> { "04", "08" };
 
+        /// <summary>
+        /// 生成所有{音节, 音频名}的字典
+        /// </summary>
+        public static void GenerateMappings()
+        {
+            foreach(var (consonant, consonantCode) in ConsonantAudioMapping)
+            {
+                foreach(var (finalCode, finals) in FinalAudioMapping)
+                {
+                    foreach(var (tone, toneCode) in ToneAudioMapping) 
+                    {
+                        string final = (finals.Length == 2 && AltTones.Contains(toneCode)) ? finals[1] : finals[0];
+                        string syllableCode = consonantCode + finalCode + toneCode;
+                        if (AbruptTones.Contains(toneCode))
+                        {
+                            if (final.EndsWith("ng"))
+                            {
+                                final = final.Substring(0, final.Length - 2);
+                            }
+                            SyllableMappings.TryAdd(consonant + final + "h" + tone, syllableCode);
+                            SyllableMappings.TryAdd(consonant + final + "k" + tone, syllableCode);
+                        } 
+                        else 
+                        {
+                            SyllableMappings.TryAdd(consonant + final + tone, syllableCode);
+                        }
+                    }
+                }
+            }
+            return;
+        }
+
+        /// <summary>
+        /// 将榕拼音节转换为对应音频文件名。
+        /// </summary>
+        /// <returns>Empty string if none is matched</returns>
+        public static string SyllableToAudio(string yngpingSyllable)
+        {   
+            if (SyllableMappings.Count == 0)
+            {
+                GenerateMappings();
+            }
+            if (SyllableMappings.ContainsKey(yngpingSyllable))
+            {
+                return SyllableMappings[yngpingSyllable];    
+            }
+            // TODO: add extra rules
+            return string.Empty;
+        }
+
         /// <summary>
         /// 将榕拼音节转换为对应音频文件名.
         /// </summary>
         /// <returns>Empty string if unsupported</returns>
-        public static string SyllableToAudio(string yngpingSyllable)
+        public static string SyllableToAudioDeprecated(string yngpingSyllable)
```

Comment:
is this still used? if not please remove

Reply:
No. I only kept it as a reference when I write the new method. Actually I think the `DestructureFinal` and `MapFinal` can be removed too.